org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/handlers/JDTLanguageServer.java

```diff
@@ -319,9 +331,72 @@ public void didChangeConfiguration(DidChangeConfigurationParams params) {
 			JavaLanguageServerPlugin.logException(e.getMessage(), e);
 		}
 		pm.registerWatchers();
+		configureFormatter();
 		logInfo(">>New configuration: " + settings);
 	}
 
+	public void configureFormatter() {
+		String formaterUrl = preferenceManager.getPreferences().getFormatterUrl();
+		Map<String, String> options = null;
+		if (formaterUrl != null) {
+			URL url = null;
+			try {
+				url = new URL(formaterUrl);
+			} catch (MalformedURLException e1) {
+				File file = new File(formaterUrl);
+				if (file.isFile()) {
+					try {
+						url = file.toURI().toURL();
+					} catch (MalformedURLException e) {
+						JavaLanguageServerPlugin.logInfo("Invalid formatter:" + formaterUrl);
+						return;
+					}
+				}
+			}
+			if (url != null) {
+				try (InputStream inputStream = url.openStream()) {
```

Comment:
how will that work behind a proxy?

Reply:
we probably need to add a timeout

Reply:
> how will that work behind a proxy?
as any URLs.
> we probably need to add a timeout
There is already the standard Java timeout.